go/src/koding/klient/machine/client/kite.go

```diff
@@ -46,10 +46,6 @@ func (kb *KiteBuilder) Ping(dynAddr DynamicAddrFunc) (machine.Status, machine.Ad
 // Build builds new kite client that will connect to machine using provided
 // address.
 func (kb *KiteBuilder) Build(ctx context.Context, addr machine.Addr) Client {
-	if _, err := kb.pool.Get(addr.Value); err != nil {
-		return NewDisconnected(ctx)
-	}
-
```

Comment:
I thought it's actually a nice optimisation, so if client is disconnected, you don't lookup the pool on each call.